gc/src/empty_trace.rs

```diff
@@ -0,0 +1,9 @@
+use std::rc::Rc;
+
+/// A marker trait for types that don't require tracing.
+/// TODO: Safety conditions
+pub unsafe trait EmptyTrace {}
+
+unsafe impl EmptyTrace for String {}
```

Comment:
Sort of a side note, but `Rc<String>` is unusual because of the [double indirection](https://rust-lang.github.io/rust-clippy/master/index.html#rc_buffer); typically `Rc<str>` or some [thin pointer equivalent](https://github.com/Wybxc/rust-string-comparison) would be preferred.